pkg/kvp/kvp.go

```diff
@@ -64,25 +67,27 @@ next:
 		}
 
 		l, err := unix.Read(kvp, asByteSlice)
-		if l != sizeOf {
-			return nil, ErrUnableToReadFromKVP
-		}
 		if err != nil {
+			if err == unix.EAGAIN || err == unix.EWOULDBLOCK {
+				continue
+			}
 			return nil, err
 		}
+		if l != sizeOf {
```

Comment:
Do you know if this bus device guarantees non-partial reads? I'm assuming it does but might be worth comparing with the daemon code.